packages/@react-spectrum/s2/src/AvatarGroup.tsx

```diff
@@ -0,0 +1,163 @@
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {Children, ReactNode, createContext, useContext, forwardRef} from 'react';
+import {DOMProps} from '@react-types/shared';
+import {getAllowedOverrides, StyleProps} from './style-utils' with {type: 'macro'};
+import {size, style} from '../style/spectrum-theme' with {type: 'macro'};
+import {ContextValue} from 'react-aria-components';
+import type {StyleString} from '../style/types';
+import {filterDOMProps, useId} from '@react-aria/utils';
+
+
+interface AvatarGroupProps extends StyleProps, DOMProps {
+  /** Avatar children of the avatar group. */
+  children: ReactNode,
+
+  /** The label for the avatar group. */
+  label?: string,
+
+  /**
+   * The size of the avatar group.
+   *
+   * @default 'M'
+   * */
+  size?: 'S' | 'M' | 'L' | 'XL' | 'XXL' | 'XXXL',
+  /* stroke color (should match the container background color) */
+
+  strokeColor?: 'base' | 'layer-1' | 'layer-2' | 'pasteboard'
+}
+
+interface AvatarGroupItemProps {
+  /** Text description of the avatar. */
+  alt?: string,
+  /** The image URL for the avatar. */
+  src: string
+}
+
+interface AvatarGroupContextProps extends Pick<AvatarGroupProps, 'strokeColor'> {
+  styles?: StyleString
+}
+
+const avatar = style({
+  marginStart: {
+    default: {
+      size: {
+        S: size(-5),
+        M: size(-6),
+        L: size(-7),
+        XL: size(-8),
+        XXL: size(-9),
+        XXXL: size(-10)
+      }
+    },
+    isFirst: 0
+  },
+  size: {
+    size: {
+      S: 20,
+      M: 24,
+      L: 28,
+      XL: 32,
+      XXL: 36,
+      XXXL: 40
+    }
+  }
+});
+
+const text = style({
+  marginStart: 8,
+  fontFamily: 'sans',
+  fontSize: {
+    size: {
+      S: 'ui-sm',
+      M: 'ui',
+      L: 'ui-lg',
+      XL: 'ui-xl',
+      XXL: 'ui-2xl',
+      XXXL: 'ui-3xl'
+    }
+  }
+});
+
+const imageStyles = style({
+  borderRadius: 'full',
+  borderWidth: 1,
+  borderStyle: 'solid',
+  '--avatarBorder': {
+    type: 'backgroundColor',
+    value: {
+      strokeColor: {
+        base: 'base',
+        'layer-1': 'layer-1',
+        'layer-2': 'layer-2',
+        pasteboard: 'pasteboard'
+      }
+    }
+  },
+  borderColor: '[var(--avatarBorder)]',
+  size: 20,
+  disableTapHighlight: true
+}, getAllowedOverrides({height: true}));
+
+const AvatarGroupContext = createContext<AvatarGroupContextProps>({});
+
+const container = style({
+  display: 'flex',
+  alignItems: 'center',
+  listStyleType: 'none',
+  paddingStart: 0,
+  margin: 0
+}, getAllowedOverrides());
+
+function AvatarGroup(props: AvatarGroupProps) {
+  let {children, label, size = 'M', strokeColor = 'base', styles, UNSAFE_style, UNSAFE_className, ...otherProps} = props;
+  let labelId = useId();
+
+  return (
+    <ul
+      aria-labelledby={labelId}
+      {...filterDOMProps(otherProps)}
+      tabIndex={0}
+      role="group"
```

Comment:
why are we using an implicit list and converting it to a group? should we just be using divs? or should their be a wrapper element that is the group, and inside is a list and a span label thing?

Reply:
ah, interesting https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/group_role
This is i think pretty specific to menu implementations

I don't know if it has further implications, would be best to consult with @majornista about this